src/main/java/io/confluent/connect/jdbc/dialect/PostgreSqlDatabaseDialect.java

```diff
@@ -362,6 +363,81 @@ protected void formatColumnValue(
     }
   }
 
+  @SuppressWarnings("unchecked")
+  @Override
+  protected boolean maybeBindPrimitive(
+      PreparedStatement statement,
+      int index,
+      Schema schema,
+      Object value
+  ) throws SQLException {
+
+    switch (schema.type()) {
+      case ARRAY: {
+        Class valueClass = value.getClass();
+        Object newValue = null;
+        if (valueClass.isArray()) {
+          newValue = value;
+        } else {
+          // All typecasts below are based on pgjdbc's documentation on how to use primitive arrays
+          // - https://jdbc.postgresql.org/documentation/head/arrays.html
+          switch (schema.valueSchema().type()) {
+            case INT8: {
+              // Gotta do this the long way, as Postgres has no single-byte integer,
+              // so we want to cast to short as the next best thing, and we can't do that with
+              // toArray.
+              Collection<Byte> listVal = (Collection<Byte>) value;
+              newValue = new short[listVal.size()];
+              int currIndex = 0;
+              for (Byte currVal : listVal) {
+                ((short[]) newValue)[currIndex++] = currVal;
+              }
+              break;
+            }
+            case INT16:
+              newValue = ArrayUtils.toPrimitive(((Collection<Short>) value)
```

Comment:
Any reason you had to convert these to unboxed primitives? The documentation you linked above says arrays of boxed primitives should also work. And, if the Object is already an array, the if clause above doesn't check whether the array contains boxed or unboxed primitives. If the unboxed version works, we could get rid of the `toPrimitive()` calls and not have to pull in commons-lang3.